Log Message not helpful: "RuntimeException: Failed to convert JSON to MessagePack"

**Describe the bug**
Observed in production error logs: https://console.cloud.google.com/errors/CPy9i6__p9PBLg?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911


**Expected behavior**
* The logged error message should contain the input string

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.RuntimeException: Failed to convert JSON to MessagePack
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:80) ~[zeebe-protocol-impl-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:71) ~[zeebe-protocol-impl-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.RequestMapper.ensureJsonSet(RequestMapper.java:157) ~[zeebe-gateway-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.RequestMapper.toCreateProcessInstanceRequest(RequestMapper.java:96) ~[zeebe-gateway-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.EndpointManager.sendRequestWithRetryPartitions(EndpointManager.java:326) [zeebe-gateway-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.EndpointManager.createProcessInstance(EndpointManager.java:161) [zeebe-gateway-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.GatewayGrpcService.createProcessInstance(GatewayGrpcService.java:75) [zeebe-gateway-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.gateway.protocol.GatewayGrpc$MethodHandlers.invoke(GatewayGrpc.java:1483) [zeebe-gateway-protocol-impl-1.0.0.jar:1.0.0]
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182) [grpc-stub-1.37.0.jar:1.37.0]
	at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35) [grpc-api-1.37.0.jar:1.37.0]
	at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23) [grpc-api-1.37.0.jar:1.37.0]
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331) [grpc-core-1.37.0.jar:1.37.0]
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:797) [grpc-core-1.37.0.jar:1.37.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) [grpc-core-1.37.0.jar:1.37.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123) [grpc-core-1.37.0.jar:1.37.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('m' (code 109)): was expecting double-quote to start field name
 at [Source: (ByteArrayInputStream); line: 1, column: 3]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2337) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:710) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:635) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleOddName(UTF8StreamJsonParser.java:2054) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._parseName(UTF8StreamJsonParser.java:1705) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:766) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.JsonGenerator._copyCurrentContents(JsonGenerator.java:2513) ~[jackson-core-2.12.3.jar:2.12.3]
	at com.fasterxml.jackson.core.JsonGenerator.copyCurrentStructure(JsonGenerator.java:2494) ~[jackson-core-2.12.3.jar:2.12.3]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convert(MsgPackConverter.java:136) ~[zeebe-protocol-impl-1.0.0.jar:1.0.0]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:76) ~[zeebe-protocol-impl-1.0.0.jar:1.0.0]
	... 17 more
```

</p>
</details>


**Environment:**
 - OS: Linux
 - Zeebe Version: 1.0.0
 - Configuration: Camunda Cloud

--- a/gateway/src/main/java/io/camunda/zeebe/gateway/RequestMapper.java
+++ b/gateway/src/main/java/io/camunda/zeebe/gateway/RequestMapper.java
@@ -7,6 +7,9 @@
  */
 package io.camunda.zeebe.gateway;
 
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import com.fasterxml.jackson.core.JsonParseException;
 import io.camunda.zeebe.gateway.impl.broker.request.BrokerActivateJobsRequest;
 import io.camunda.zeebe.gateway.impl.broker.request.BrokerCancelProcessInstanceRequest;
 import io.camunda.zeebe.gateway.impl.broker.request.BrokerCompleteJobRequest;
@@ -150,11 +153,30 @@ public final class RequestMapper {
     return new BrokerResolveIncidentRequest(grpcRequest.getIncidentKey());
   }
 
-  private static DirectBuffer ensureJsonSet(final String value) {
+  static DirectBuffer ensureJsonSet(final String value) {
     if (value == null || value.trim().isEmpty()) {
       return DocumentValue.EMPTY_DOCUMENT;
     } else {
-      return new UnsafeBuffer(MsgPackConverter.convertToMsgPack(value));
+      try {
+        return new UnsafeBuffer(MsgPackConverter.convertToMsgPack(value));
+      } catch (final RuntimeException e) {
+        final var cause = e.getCause();
+        if (cause instanceof JsonParseException) {
+          final var parseException = (JsonParseException) cause;
+
+          final var descriptiveException =
+              new JsonParseException(
+                  parseException.getProcessor(),
+                  "Invalid JSON value: " + value,
+                  parseException.getLocation(),
+                  cause);
+
+          rethrowUnchecked(descriptiveException);
+          return DocumentValue.EMPTY_DOCUMENT; // bogus return statement
+        } else {
+          throw e;
+        }
+      }
     }
   }
 }

--- a/gateway/src/test/java/io/camunda/zeebe/gateway/RequestMapperTest.java
+++ b/gateway/src/test/java/io/camunda/zeebe/gateway/RequestMapperTest.java
@@ -1,0 +1,30 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import org.junit.jupiter.api.Test;
+
+public class RequestMapperTest {
+
+  @Test
+  public void shouldThrowHelpfulExceptionIfJsonIsInvalid() {
+    // given
+    final var invalidJson = "{ \"test\": \"value\", \"error\": \"errorrvalue }";
+    // closing quote missing in second value
+
+    // when + then
+    assertThatThrownBy(() -> RequestMapper.ensureJsonSet(invalidJson))
+        .isInstanceOf(JsonParseException.class)
+        .hasMessageContaining("Invalid JSON", invalidJson)
+        .getCause()
+        .isInstanceOf(JsonParseException.class);
+  }
+}

